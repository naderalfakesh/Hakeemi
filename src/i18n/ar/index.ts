import common from './common.json';
import home from './home.json';
import signIn from './signIn.json';
import signup from './signup.json';
import welcome from './welcome.json';

export default {
  common,
  home,
  signIn,
  signup,
  welcome,
};
