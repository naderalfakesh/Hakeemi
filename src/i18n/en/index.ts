import common from './common.json';
import home from './home.json';
import login from './login.json';
import signup from './signup.json';
import welcome from './welcome.json';

export default {
  common,
  home,
  login,
  signup,
  welcome,
};
