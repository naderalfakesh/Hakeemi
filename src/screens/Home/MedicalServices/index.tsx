import MedicalServices from './MedicalServices';

export default MedicalServices;
