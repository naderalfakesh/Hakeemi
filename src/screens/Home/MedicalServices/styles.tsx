import { StyleSheet } from 'react-native';
import spacing from '../../../theme/spacing';

export default StyleSheet.create({
  base: {
    flexDirection: 'row',
    flexWrap: 'wrap',
    justifyContent: 'space-between',
  },
  button: {
    marginBottom: spacing.small1,
    marginRight: spacing.small4,
  },
  submitButton: {
    marginVertical: spacing.small1,
  },
});
