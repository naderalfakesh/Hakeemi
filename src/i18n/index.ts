import i18n from './i18n';

export * from './i18n';

export default i18n;
