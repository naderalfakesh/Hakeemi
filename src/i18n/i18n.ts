import i18next from 'i18next';
import en from './en';
import ar from './ar';

const i18n = i18next.createInstance();
i18n.init({
  fallbackLng: 'en',
  lng: 'en',
  supportedLngs: ['en', 'ar'],
  defaultNS: 'common',
  resources: {
    en,
    ar,
  },
  react: {
    useSuspense: false,
  },
});

export default i18n;
