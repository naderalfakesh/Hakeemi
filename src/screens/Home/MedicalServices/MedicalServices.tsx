import React, { Fragment, useState } from 'react';
import { useTranslation } from 'react-i18next';
import { View, ViewStyle } from 'react-native';
import Button from '../../../components/Button';
import { IconName } from '../../../components/Icon';
import styles from './styles';

const SERVICES: {
  id: string;
  iconName: IconName;
}[] = [
  { id: '1', iconName: 'diagnose' },
  { id: '2', iconName: 'badge' },
  { id: '3', iconName: 'shot' },
  { id: '4', iconName: 'ambulance' },
  { id: '5', iconName: 'heart-beat' },
  { id: '6', iconName: 'checkup' },
  { id: '7', iconName: 'pill' },
  { id: '8', iconName: 'heart-plus' },
];

interface Props {
  style?: ViewStyle;
  onPress?: (services: Array<string>) => void;
}

const MedicalServices = ({ style, onPress }: Props) => {
  const { t } = useTranslation('home');
  const [selected, setSelected] = useState<Array<string>>([]);

  return (
    <Fragment>
      <View style={[styles.base, style]}>
        {SERVICES.map(({ id, iconName }, index) => (
          <Button
            key={id}
            style={(index + 1) % 4 ? styles.button : undefined}
            highlight={false}
            theme={selected.includes(id) ? 'primary' : 'grey'}
            size="squareIcon"
            iconName={iconName}
            onPress={() =>
              selected.includes(id)
                ? setSelected(selected.filter(item => item !== id))
                : setSelected([...selected, id])
            }>
            {t(('services.types.' + iconName) as any)}
          </Button>
        ))}
      </View>
      <Button
        testID="search-button"
        onPress={() => onPress && onPress(selected)}
        style={styles.submitButton}>
        {t('services.button')}
      </Button>
    </Fragment>
  );
};

export default MedicalServices;
