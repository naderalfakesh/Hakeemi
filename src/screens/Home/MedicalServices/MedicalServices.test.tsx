import React, { FC } from 'react';
import { render, fireEvent } from '@testing-library/react-native';
import MedicalServices from './MedicalServices';
import { ReactTestInstance } from 'react-test-renderer';
import { StyleSheet } from 'react-native';
import { I18nextProvider } from 'react-i18next';
import i18n from '../../../i18n';

const wrapper: FC = ({ children }) => {
  return <I18nextProvider i18n={i18n}>{children}</I18nextProvider>;
};
describe('Button component test', () => {
  it('renders correctly', () => {
    const { toJSON } = render(<MedicalServices />, { wrapper });
    expect(toJSON()).toMatchSnapshot();
  });

  it('overrides styles', () => {
    const { container } = render(<MedicalServices style={{ padding: 50 }} />);
    const baseView = StyleSheet.flatten(
      (container.children[0] as ReactTestInstance).props.style,
    );
    expect(baseView.padding).toBe(50);
  });

  it('invokes a function when close icon is pressed', () => {
    const mockFn = jest.fn();
    const { getByTestId } = render(<MedicalServices onPress={mockFn} />);
    const btn = getByTestId('search-button');
    fireEvent.press(btn);
    expect(mockFn).toBeCalled();
  });
});
