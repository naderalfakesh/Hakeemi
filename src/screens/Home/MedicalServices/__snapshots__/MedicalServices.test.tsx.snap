// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Button component test renders correctly 1`] = `
Array [
  <View
    style={
      Array [
        Object {
          "flexDirection": "row",
          "flexWrap": "wrap",
          "justifyContent": "space-between",
        },
        undefined,
      ]
    }
  >
    <View
      accessible={true}
      collapsable={false}
      focusable={true}
      onBlur={[Function]}
      onClick={[Function]}
      onFocus={[Function]}
      onResponderGrant={[Function]}
      onResponderMove={[Function]}
      onResponderRelease={[Function]}
      onResponderTerminate={[Function]}
      onResponderTerminationRequest={[Function]}
      onStartShouldSetResponder={[Function]}
      style={
        Array [
          Object {
            "alignItems": "center",
            "borderRadius": 16,
            "justifyContent": "center",
          },
          Object {
            "backgroundColor": "#f2f7f9",
          },
          Object {
            "height": 70,
            "padding": 10,
            "width": 75,
          },
          false,
          false,
          Object {
            "marginBottom": 10,
            "marginRight": 18,
          },
        ]
      }
      testID="Button"
    >
      <Text
        allowFontScaling={false}
        style={
          Array [
            Object {
              "color": "#00bbd4",
              "fontSize": 24,
            },
            Array [
              Object {
                "lineHeight": 24,
                "textAlign": "center",
              },
              Object {
                "padding": 6,
              },
            ],
            Object {
              "fontFamily": "doctor-font",
              "fontStyle": "normal",
              "fontWeight": "normal",
            },
            Object {},
          ]
        }
      >
        
      </Text>
      <Text
        style={
          Object {
            "color": "#a6aaad",
            "fontFamily": "System",
            "fontSize": 10,
            "fontWeight": "400",
            "includeFontPadding": false,
            "lineHeight": 10,
            "textAlign": undefined,
          }
        }
        testID="Text"
      >
        Diagnosis
      </Text>
    </View>
    <View
      accessible={true}
      collapsable={false}
      focusable={true}
      onBlur={[Function]}
      onClick={[Function]}
      onFocus={[Function]}
      onResponderGrant={[Function]}
      onResponderMove={[Function]}
      onResponderRelease={[Function]}
      onResponderTerminate={[Function]}
      onResponderTerminationRequest={[Function]}
      onStartShouldSetResponder={[Function]}
      style={
        Array [
          Object {
            "alignItems": "center",
            "borderRadius": 16,
            "justifyContent": "center",
          },
          Object {
            "backgroundColor": "#f2f7f9",
          },
          Object {
            "height": 70,
            "padding": 10,
            "width": 75,
          },
          false,
          false,
          Object {
            "marginBottom": 10,
            "marginRight": 18,
          },
        ]
      }
      testID="Button"
    >
      <Text
        allowFontScaling={false}
        style={
          Array [
            Object {
              "color": "#00bbd4",
              "fontSize": 24,
            },
            Array [
              Object {
                "lineHeight": 24,
                "textAlign": "center",
              },
              Object {
                "padding": 6,
              },
            ],
            Object {
              "fontFamily": "doctor-font",
              "fontStyle": "normal",
              "fontWeight": "normal",
            },
            Object {},
          ]
        }
      >
        
      </Text>
      <Text
        style={
          Object {
            "color": "#a6aaad",
            "fontFamily": "System",
            "fontSize": 10,
            "fontWeight": "400",
            "includeFontPadding": false,
            "lineHeight": 10,
            "textAlign": undefined,
          }
        }
        testID="Text"
      >
        Bandage
      </Text>
    </View>
    <View
      accessible={true}
      collapsable={false}
      focusable={true}
      onBlur={[Function]}
      onClick={[Function]}
      onFocus={[Function]}
      onResponderGrant={[Function]}
      onResponderMove={[Function]}
      onResponderRelease={[Function]}
      onResponderTerminate={[Function]}
      onResponderTerminationRequest={[Function]}
      onStartShouldSetResponder={[Function]}
      style={
        Array [
          Object {
            "alignItems": "center",
            "borderRadius": 16,
            "justifyContent": "center",
          },
          Object {
            "backgroundColor": "#f2f7f9",
          },
          Object {
            "height": 70,
            "padding": 10,
            "width": 75,
          },
          false,
          false,
          Object {
            "marginBottom": 10,
            "marginRight": 18,
          },
        ]
      }
      testID="Button"
    >
      <Text
        allowFontScaling={false}
        style={
          Array [
            Object {
              "color": "#00bbd4",
              "fontSize": 24,
            },
            Array [
              Object {
                "lineHeight": 24,
                "textAlign": "center",
              },
              Object {
                "padding": 6,
              },
            ],
            Object {
              "fontFamily": "doctor-font",
              "fontStyle": "normal",
              "fontWeight": "normal",
            },
            Object {},
          ]
        }
      >
        
      </Text>
      <Text
        style={
          Object {
            "color": "#a6aaad",
            "fontFamily": "System",
            "fontSize": 10,
            "fontWeight": "400",
            "includeFontPadding": false,
            "lineHeight": 10,
            "textAlign": undefined,
          }
        }
        testID="Text"
      >
        Shots
      </Text>
    </View>
    <View
      accessible={true}
      collapsable={false}
      focusable={true}
      onBlur={[Function]}
      onClick={[Function]}
      onFocus={[Function]}
      onResponderGrant={[Function]}
      onResponderMove={[Function]}
      onResponderRelease={[Function]}
      onResponderTerminate={[Function]}
      onResponderTerminationRequest={[Function]}
      onStartShouldSetResponder={[Function]}
      style={
        Array [
          Object {
            "alignItems": "center",
            "borderRadius": 16,
            "justifyContent": "center",
          },
          Object {
            "backgroundColor": "#f2f7f9",
          },
          Object {
            "height": 70,
            "padding": 10,
            "width": 75,
          },
          false,
          false,
          undefined,
        ]
      }
      testID="Button"
    >
      <Text
        allowFontScaling={false}
        style={
          Array [
            Object {
              "color": "#00bbd4",
              "fontSize": 24,
            },
            Array [
              Object {
                "lineHeight": 24,
                "textAlign": "center",
              },
              Object {
                "padding": 6,
              },
            ],
            Object {
              "fontFamily": "doctor-font",
              "fontStyle": "normal",
              "fontWeight": "normal",
            },
            Object {},
          ]
        }
      >
        
      </Text>
      <Text
        style={
          Object {
            "color": "#a6aaad",
            "fontFamily": "System",
            "fontSize": 10,
            "fontWeight": "400",
            "includeFontPadding": false,
            "lineHeight": 10,
            "textAlign": undefined,
          }
        }
        testID="Text"
      >
        Ambulance
      </Text>
    </View>
    <View
      accessible={true}
      collapsable={false}
      focusable={true}
      onBlur={[Function]}
      onClick={[Function]}
      onFocus={[Function]}
      onResponderGrant={[Function]}
      onResponderMove={[Function]}
      onResponderRelease={[Function]}
      onResponderTerminate={[Function]}
      onResponderTerminationRequest={[Function]}
      onStartShouldSetResponder={[Function]}
      style={
        Array [
          Object {
            "alignItems": "center",
            "borderRadius": 16,
            "justifyContent": "center",
          },
          Object {
            "backgroundColor": "#f2f7f9",
          },
          Object {
            "height": 70,
            "padding": 10,
            "width": 75,
          },
          false,
          false,
          Object {
            "marginBottom": 10,
            "marginRight": 18,
          },
        ]
      }
      testID="Button"
    >
      <Text
        allowFontScaling={false}
        style={
          Array [
            Object {
              "color": "#00bbd4",
              "fontSize": 24,
            },
            Array [
              Object {
                "lineHeight": 24,
                "textAlign": "center",
              },
              Object {
                "padding": 6,
              },
            ],
            Object {
              "fontFamily": "doctor-font",
              "fontStyle": "normal",
              "fontWeight": "normal",
            },
            Object {},
          ]
        }
      >
        
      </Text>
      <Text
        style={
          Object {
            "color": "#a6aaad",
            "fontFamily": "System",
            "fontSize": 10,
            "fontWeight": "400",
            "includeFontPadding": false,
            "lineHeight": 10,
            "textAlign": undefined,
          }
        }
        testID="Text"
      >
        Heartbeats
      </Text>
    </View>
    <View
      accessible={true}
      collapsable={false}
      focusable={true}
      onBlur={[Function]}
      onClick={[Function]}
      onFocus={[Function]}
      onResponderGrant={[Function]}
      onResponderMove={[Function]}
      onResponderRelease={[Function]}
      onResponderTerminate={[Function]}
      onResponderTerminationRequest={[Function]}
      onStartShouldSetResponder={[Function]}
      style={
        Array [
          Object {
            "alignItems": "center",
            "borderRadius": 16,
            "justifyContent": "center",
          },
          Object {
            "backgroundColor": "#f2f7f9",
          },
          Object {
            "height": 70,
            "padding": 10,
            "width": 75,
          },
          false,
          false,
          Object {
            "marginBottom": 10,
            "marginRight": 18,
          },
        ]
      }
      testID="Button"
    >
      <Text
        allowFontScaling={false}
        style={
          Array [
            Object {
              "color": "#00bbd4",
              "fontSize": 24,
            },
            Array [
              Object {
                "lineHeight": 24,
                "textAlign": "center",
              },
              Object {
                "padding": 6,
              },
            ],
            Object {
              "fontFamily": "doctor-font",
              "fontStyle": "normal",
              "fontWeight": "normal",
            },
            Object {},
          ]
        }
      >
        
      </Text>
      <Text
        style={
          Object {
            "color": "#a6aaad",
            "fontFamily": "System",
            "fontSize": 10,
            "fontWeight": "400",
            "includeFontPadding": false,
            "lineHeight": 10,
            "textAlign": undefined,
          }
        }
        testID="Text"
      >
        Checkup
      </Text>
    </View>
    <View
      accessible={true}
      collapsable={false}
      focusable={true}
      onBlur={[Function]}
      onClick={[Function]}
      onFocus={[Function]}
      onResponderGrant={[Function]}
      onResponderMove={[Function]}
      onResponderRelease={[Function]}
      onResponderTerminate={[Function]}
      onResponderTerminationRequest={[Function]}
      onStartShouldSetResponder={[Function]}
      style={
        Array [
          Object {
            "alignItems": "center",
            "borderRadius": 16,
            "justifyContent": "center",
          },
          Object {
            "backgroundColor": "#f2f7f9",
          },
          Object {
            "height": 70,
            "padding": 10,
            "width": 75,
          },
          false,
          false,
          Object {
            "marginBottom": 10,
            "marginRight": 18,
          },
        ]
      }
      testID="Button"
    >
      <Text
        allowFontScaling={false}
        style={
          Array [
            Object {
              "color": "#00bbd4",
              "fontSize": 24,
            },
            Array [
              Object {
                "lineHeight": 24,
                "textAlign": "center",
              },
              Object {
                "padding": 6,
              },
            ],
            Object {
              "fontFamily": "doctor-font",
              "fontStyle": "normal",
              "fontWeight": "normal",
            },
            Object {},
          ]
        }
      >
        
      </Text>
      <Text
        style={
          Object {
            "color": "#a6aaad",
            "fontFamily": "System",
            "fontSize": 10,
            "fontWeight": "400",
            "includeFontPadding": false,
            "lineHeight": 10,
            "textAlign": undefined,
          }
        }
        testID="Text"
      >
        Medicines
      </Text>
    </View>
    <View
      accessible={true}
      collapsable={false}
      focusable={true}
      onBlur={[Function]}
      onClick={[Function]}
      onFocus={[Function]}
      onResponderGrant={[Function]}
      onResponderMove={[Function]}
      onResponderRelease={[Function]}
      onResponderTerminate={[Function]}
      onResponderTerminationRequest={[Function]}
      onStartShouldSetResponder={[Function]}
      style={
        Array [
          Object {
            "alignItems": "center",
            "borderRadius": 16,
            "justifyContent": "center",
          },
          Object {
            "backgroundColor": "#f2f7f9",
          },
          Object {
            "height": 70,
            "padding": 10,
            "width": 75,
          },
          false,
          false,
          undefined,
        ]
      }
      testID="Button"
    >
      <Text
        allowFontScaling={false}
        style={
          Array [
            Object {
              "color": "#00bbd4",
              "fontSize": 24,
            },
            Array [
              Object {
                "lineHeight": 24,
                "textAlign": "center",
              },
              Object {
                "padding": 6,
              },
            ],
            Object {
              "fontFamily": "doctor-font",
              "fontStyle": "normal",
              "fontWeight": "normal",
            },
            Object {},
          ]
        }
      >
        
      </Text>
      <Text
        style={
          Object {
            "color": "#a6aaad",
            "fontFamily": "System",
            "fontSize": 10,
            "fontWeight": "400",
            "includeFontPadding": false,
            "lineHeight": 10,
            "textAlign": undefined,
          }
        }
        testID="Text"
      >
        Advice
      </Text>
    </View>
  </View>,
  <View
    accessible={true}
    collapsable={false}
    focusable={true}
    onBlur={[Function]}
    onClick={[Function]}
    onFocus={[Function]}
    onResponderGrant={[Function]}
    onResponderMove={[Function]}
    onResponderRelease={[Function]}
    onResponderTerminate={[Function]}
    onResponderTerminationRequest={[Function]}
    onStartShouldSetResponder={[Function]}
    style={
      Array [
        Object {
          "alignItems": "center",
          "borderRadius": 16,
          "justifyContent": "center",
        },
        Object {
          "backgroundColor": "#00bbd4",
        },
        Object {
          "padding": 16,
        },
        false,
        false,
        Object {
          "marginVertical": 10,
        },
      ]
    }
    testID="search-button"
  >
    <Text
      style={
        Object {
          "color": "#ffffff",
          "fontFamily": "System",
          "fontSize": 16,
          "fontWeight": "600",
          "includeFontPadding": false,
          "lineHeight": 20,
          "textAlign": undefined,
        }
      }
      testID="Text"
    >
      Available doctors
    </Text>
  </View>,
]
`;
